packages/react-router/lib/router/router.ts

```diff
@@ -276,7 +276,7 @@ export interface RouterState {
   /**
    * The action of the most recent navigation
    */
-  historyAction: HistoryAction;
+  historyAction: NavigationType;
```

Comment:
I wonder if we should rename this to `navigationType` in v7 to align with `useNavigationType`...?